driver-core/src/main/java/com/datastax/driver/core/policies/DefaultRetryPolicy.java

```diff
@@ -135,4 +135,17 @@ public RetryDecision onUnavailable(Statement statement, ConsistencyLevel cl, int
             ? RetryDecision.tryNextHost(cl)
             : RetryDecision.rethrow();
     }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This implementation triggers a retry on the next host in the query plan,
+     * regardless of the consistency level, the number of retries or
+     * the possibility that the mutation has been applied server-side.
+     */
+    @Override
+    public RetryDecision onUnexpectedError(Statement statement, ConsistencyLevel cl, int nbRetry, boolean mightHaveBeenApplied) {
```

Comment:
The only issue i have with not having the exception is you don't have any idea why there was an error, so retrying may not be the right behavior depending on the criticality of the error.  Some delineation of why there was an error might be nice.   I just noticed [this discussion](https://github.com/datastax/java-driver/pull/538/files#r45959031) so will carry on there with my input.

I agree 'unexpected' might be perceived as too generic.  +1 for `onRequestError`.  Straight up `onError` might be good as well, since we can assume that this is